### PR TITLE
Job env variables

### LINKED ISSUE #338: Job ENV variables are shared between init container & running container
**Describe the bug:** (a clear and concise description of what the bug is)
There is one config map created for both the init container and the running container. When you have the same values in the init as well as running container, it would store the value twice in the config map, making otomi/k8s complain

**To Reproduce**
Create a new Job:
```yaml
{{- $v := .Environment.Values }}
{{- $c := $v.charts }}

type: Job
name: gitea
enabled: true
env:
  GITEA_URL: https://gitea.{{ $v.cluster.domain }}
  GITEA_PASSWORD: {{ $c | get "gitea.admin.password" "bladibla"}}
init:
  env:
    GITEA_URL: https://gitea.{{ $v.cluster.domain }}
    GITEA_PASSWORD: {{ $c | get "gitea.admin.password" "bladibla"}}
  image:
    repository: otomi/tools
    tag: v1.4.13
    pullPolicy: IfNotPresent
  shell: bash
  script: |
    {{- readFile "scripts/init/gitea.sh" | nindent 4 }}
image:
  repository: otomi/tools
  tag: v1.4.13
  pullPolicy: IfNotPresent
shell: bash
script: |
  {{- readFile "scripts/gitea.sh" | nindent 2 }}
runPolicy: Always
```

The config map created would be templated as follows:
```yaml
# Source: jobs/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: job-gitea
  labels:
    helm.sh/chart: jobs-0.1.0
    app.kubernetes.io/name: jobs
    app.kubernetes.io/instance: job-gitea
    app.kubernetes.io/version: "0.1.0"
    app.kubernetes.io/managed-by: Helm
data:
    GITEA_PASSWORD: bladibla
    GITEA_URL: https://gitea.dev.eks.otomi.cloud
    GITEA_PASSWORD: bladibla
    GITEA_URL: https://gitea.dev.eks.otomi.cloud
---
``` 
Where data contains duplicate keys (and values)

**Expected behavior:** (a clear and concise description of what you expected to happen)
Not storing duplicate key/value pairs.

**Versions:** (seen in console's `/settings`)
- Core Version - current master

**Proposed solution:**
Create two config maps, one for the init container, and one for the running container.
Would need to double check that the existing jobs move the ENV variables accordingly.

### PR DESCRIPTION
**Pull request by Dunky13**
Fixes #338 

Create a separate init configmap, so env variables aren't shared and don't cause collision.

### CODE REVIEW

File: charts/jobs/templates/configmap.yaml [hunk 1]
```diff
@@ -9,6 +9,15 @@ data:
   {{- with $v.env }}
     {{- toYaml . | nindent 4 }}
   {{- end }}
```
Comment (Morriz) [inline charts/jobs/templates/configmap.yaml:11]:
```suggestion
  {{- toYaml $v.env | nindent 4 }}
```

File: charts/jobs/templates/configmap.yaml [hunk 2]
```diff
@@ -9,6 +9,15 @@ data:
...
+{{- if $v.init.env }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: job-{{ $v.name }}-name
+  labels: {{- include "jobs.labels" $ | nindent 4 }}
+data:
   {{- with $v.init.env  }}
     {{- toYaml . | nindent 4 }}
   {{- end }}
```
Comment (Morriz) [inline charts/jobs/templates/configmap.yaml:23]:
```suggestion
  {{- toYaml $v.init.env | nindent 4 }}
```

**Review by Morriz — CHANGES_REQUESTED** [review]

File: charts/jobs/templates/job.yaml [hunk 3]
```diff
@@ -37,7 +37,7 @@ spec:
         {{- if $v.init.env }}
         envFrom:
         - configMapRef:
-            name: job-{{ $v.name }}
+            name: job-{{ $v.name }}-init
```
Comment (svatwork) [inline charts/jobs/templates/job.yaml:40]: How are you referring to this `configMap`? I don't see any with this name in `configmap.yaml`.

Comment (Dunky13) [reply charts/jobs/templates/job.yaml:40]: https://github.com/redkubes/otomi-core/pull/408/files#diff-18dafd376c0c31873988c49de11a03e36b66f770514ee98c7794c267ea72f83bR18

Comment (Dunky13) [reply charts/jobs/templates/job.yaml:40]: It is there on line 16.

**Review by Morriz — CHANGES_REQUESTED** [review]
Did you just fix this in the mean time already? Just saw a notification flashing

File: charts/jobs/templates/configmap.yaml [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
   name: job-{{ $v.name }}
   labels: {{- include "jobs.labels" $ | nindent 4 }}
 data:
-  {{- with $v.env }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
-  {{- with $v.init.env  }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+  {{- toYaml ($v | get "env" dict) | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (Dunky13) [reply charts/jobs/templates/configmap.yaml]: I fixed this already

File: charts/jobs/templates/configmap.yaml [hunk 5]
```diff
@@ -1,17 +1,22 @@
 {{- $v := .Values }}
-{{- if or (or $v.env $v.init.env) $v.files }}
+{{- if $v.env }}
```
[the remark this answers is not in the document]
Comment (Dunky13) [reply charts/jobs/templates/configmap.yaml:2]: So the if here is correct?

File: charts/jobs/templates/job.yaml (hunk 3 above)
Comment (svatwork) [reply charts/jobs/templates/job.yaml:40]: It refers to `job-{{ $v.name }}-name` on L16, and here the configMapRef points to `job-{{ $v.name }}-init`. Am I missing something?

File: charts/jobs/templates/configmap.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Morriz) [reply charts/jobs/templates/configmap.yaml:2]: Excuse my verbosity

File: charts/jobs/templates/job.yaml (hunk 3 above)
Comment (Dunky13) [reply charts/jobs/templates/job.yaml:40]: You missed giving me a wakeup slap ;-) Thanks for noticing!

**Review by Morriz — APPROVED** [review]

**Pull request merged**